torch/csrc/utils/python_numbers.h

```diff
@@ -132,3 +132,24 @@ inline std::complex<double> THPUtils_unpackComplexDouble(PyObject *obj) {
 
   return std::complex<double>(value.real, value.imag);
 }
+
+inline bool THPUtils_unpackNumberAsBool(PyObject* obj) {
+  //when obj is float(contains NaN, inf, -inf)
+  if (PyFloat_Check(obj)) {
+    return (bool)PyFloat_AS_DOUBLE(obj);
+  }
+
+  //when obj is complex number
+  if (PyComplex_Check(obj)) {
+    double real_val = PyComplex_RealAsDouble(obj);
+    double imag_val = PyComplex_ImagAsDouble(obj);
+    return !(real_val == 0 && imag_val == 0);
+  }
+  //when obj is other
+  int overflow;
```

Comment:
You're acquiring the overflow but not checking it. Do you need a check like:

```
if (overflow != 0) {
  throw std::runtime_error("Overflow when unpacking double");
}
```
?

Reply:
When the value is overflow(for example: torch.iinfo(torch.int64).max + 1), it will return True in numpy and python, so I deleted the overflow check ensure it can also return True at this time.

Reply:
Cool! Would you add a comment explaining this?

Reply:
Sure! I‘ll add a coment.